tests/Unit/Utilities/PregTest.php

```diff
@@ -0,0 +1,219 @@
+        @$subject->replace('/', '', '');
+    }
+
+    /**
+     * @test
+     */
+    public function exceptionThrownAfterThrowExceptionsTurnedOn(): void
+    {
+        $this->expectException(\RuntimeException::class);
+        $subject = new Preg();
+
+        $subject->throwExceptions(true);
+        @$subject->replace('/', '', '');
+    }
+
+    /**
+     * @test
+     */
+    public function exceptionNotThrownAfterThrowExceptionsTurnedOff(): void
+    {
+        $this->expectNotToPerformAssertions();
+        $subject = new Preg();
+
+        $subject->throwExceptions(true);
+        $subject->throwExceptions(false);
+        @$subject->replace('/', '', '');
+    }
+
+    /**
+     * @return array<array<array-key, string|non-empty-array<string>|int>>
+     */
+    public function providePregReplaceArgumentsAndExpectedResult(): array
+    {
+        return [
+            'string arguments' => [
+                'pattern' => '/a/',
+                'replacement' => 'fab',
+                'subject' => 'abba',
+                'limit' => -1,
+                'expect' => 'fabbbfab',
+            ],
+            'array pattern' => [
+                'pattern' => ['/a/', '/b/'],
+                'replacement' => 'fab',
+                'subject' => 'abba',
+                'limit' => -1,
+                'expect' => 'fafabfabfabfafab',
+            ],
+            'array pattern and replacement' => [
+                'pattern' => ['/a/', '/b/'],
+                'replacement' => ['fab', 'z'],
+                'subject' => 'abba',
+                'limit' => -1,
+                'expect' => 'fazzzfaz',
+            ],
+            'with limit' => [
+                'pattern' => '/a/',
+                'replacement' => 'fab',
+                'subject' => 'abba',
+                'limit' => 1,
+                'expect' => 'fabbba',
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param non-empty-string|non-empty-array<non-empty-string> $pattern
+     * @param string|non-empty-array<string> $replacement
+     *
+     * @dataProvider providePregReplaceArgumentsAndExpectedResult
+     */
+    public function replaceReplaces($pattern, $replacement, string $subject, int $limit, string $expectedResult): void
+    {
+        $testSubject = new Preg();
+
+        $result = $testSubject->replace($pattern, $replacement, $subject, $limit);
+
+        self::assertSame($expectedResult, $result);
+    }
+
+    /**
+     * @test
+     *
+     * @param non-empty-string|non-empty-array<non-empty-string> $pattern
+     * @param string|non-empty-array<string> $replacement
+     *
+     * @dataProvider providePregReplaceArgumentsAndExpectedResult
+     */
+    public function replaceCallbackReplaces(
+        $pattern,
+        $replacement,
+        string $subject,
+        int $limit,
+        string $expectedResult
+    ): void {
+        $callback = static function (array $matches) use ($replacement): string {
```

Comment:
I'd strongly prefer this static function to be a class method in order to make it easier to grok. Then we could also use class properties instead of static local variables.

Reply:
Done.